src/Microsoft.ML.Data/Transforms/ValueMappingTransformer.cs

```diff
@@ -0,0 +1,995 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Data.IO;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Transforms;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+[assembly: LoadableClass(ValueMappingTransformer.Summary, typeof(IDataTransform), typeof(ValueMappingTransformer),
+    typeof(ValueMappingTransformer.Arguments), typeof(SignatureDataTransform),
+    ValueMappingTransformer.UserName, "ValueMapping", "ValueMappingTransformer", ValueMappingTransformer.ShortName, "TermLookup",
+    DocName = "transform/ValueMappingTransformer.md")]
+
+[assembly: LoadableClass(ValueMappingTransformer.Summary, typeof(IDataTransform), typeof(ValueMappingTransformer), null, typeof(SignatureLoadDataTransform),
+    "Value Mapping Transform", ValueMappingTransformer.LoaderSignature)]
+
+[assembly: LoadableClass(ValueMappingTransformer.Summary, typeof(ValueMappingTransformer), null, typeof(SignatureLoadModel),
+    "Value Mapping Transform", ValueMappingTransformer.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(IRowMapper), typeof(ValueMappingTransformer), null, typeof(SignatureLoadRowMapper),
+    ValueMappingTransformer.UserName, ValueMappingTransformer.LoaderSignature)]
+
+[assembly: LoadableClass("", typeof(IDataTransform), typeof(ValueMappingTransformer), null, typeof(SignatureLoadDataTransform),
+    "", ValueMappingTransformer.TermLookupLoaderSignature)]
+
+namespace Microsoft.ML.Transforms
+{
+    /// <summary>
+    /// The ValueMappingEstimator is a 1-1 mapping from a key to value. The key type and value type are specified
+    /// through TKey and TValue. Arrays are supported for vector types which can be used as either a key or a value
+    /// or both. The mapping is specified, not trained by providiing a list of keys and a list of values.
+    /// </summary>
+    /// <typeparam name="TKey">Specifies the key type</typeparam>
+    /// <typeparam name="TValue">Specifies the value type</typeparam>
+    public sealed class ValueMappingEstimator<TKey, TValue> : TrivialEstimator<ValueMappingTransformer<TKey, TValue>>
+    {
+        private (string input, string output)[] _columns;
+
+        /// <summary>
+        /// Constructs the ValueMappingEstimator, key type -> value type mapping
+        /// </summary>
+        /// <param name="env">Instance of the host environment</param>
+        /// <param name="keys">The list of keys of TKey</param>
+        /// <param name="values">The list of values of TValue</param>
+        /// <param name="columns">The list of columns to apply</param>
```

Comment:
nit: can they end in '.' so they appear like proper sentences in the docs website.